test/qtism/data/storage/xml/marshalling/OutcomeDeclarationMarshallerTest.php

```diff
@@ -15,6 +16,40 @@
 
 class OutcomeDeclarationMarshallerTest extends QtiSmTestCase {
 
+    public function testUnmarszallExternalScore()
+    {
+        $dom = new DOMDocument('1.0', 'UTF-8');
+        $dom->loadXML(
+            '<outcomeDeclaration xmlns="http://www.imsglobal.org/xsd/imsqti_v2p2" identifier="outcomeDeclarationRec" cardinality="record" externalScored="human"/>'
+        );
+        $element = $dom->documentElement;
+        $marshaller = $this->getMarshallerFactory()->createMarshaller($element);
+        $component = $marshaller->unmarshall($element);
+
+        $this->assertInstanceOf('qtism\\data\\state\\OutcomeDeclaration', $component);
+        $this->assertEquals($component->getExternalScore(), 'human');
+
+    }
+    public function testMarshallExternalScore()
+    {
+        // Initialize a minimal outcomeDeclaration.
+        $identifier = "outcome1";
+        $cardinality = Cardinality::SINGLE;
+        $baseType = BaseType::INTEGER;
+        $externalScore = ExternalScore::HUMAN;
```

Comment:
As said earlier, we should test the different authorized values and some not in the list.
I have to check about case sensitivity of arguments though.